app/src/main/java/com/android/sample/model/ingredient/Ingredient.kt

```diff
@@ -1,3 +1,16 @@
 package com.android.sample.model.ingredient
 
-class Ingredient(val barCode: Long, val name: String) {}
+data class Ingredient(
+    val barCode: Long? = null,
+    val name: String? = null,
```

Comment:
The barcode and name shouldn't be able to be null, as we need something to Identify them (the barcode) and we don't want to have unnamed ingredients in the database.